frame/support/procedural/src/pallet/parse/helper.rs

```diff
@@ -596,3 +598,34 @@ pub fn check_type_value_gen(
 
 	Ok(i)
 }
+
+/// Check the keyword `DispatchResultWithPostInfo` or `DispatchResult`.
+pub fn check_pallet_call_return_type(
+	type_: &syn::Type,
+) -> syn::Result<()> {
+	let expected = "expected `DispatchResultWithPostInfo` or `DispatchResult`";
+	pub struct Checker;
+	impl syn::parse::Parse for Checker {
+		fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+			let lookahead = input.lookahead1();
+			if lookahead.peek(keyword::DispatchResultWithPostInfo) {
+				input.parse::<keyword::DispatchResultWithPostInfo>()?;
+				Ok(Self)
+			} else if lookahead.peek(keyword::DispatchResult) {
+				input.parse::<keyword::DispatchResult>()?;
+				Ok(Self)
+			} else {
+				Err(lookahead.error())
+			}
+		}
+	}
+
+	syn::parse2::<Checker>(type_.to_token_stream())
+		.map_err(|e| {
+			let msg = format!("Invalid pallet::call return type: {}", expected);
```

Comment:
Why is `expected` a variable here?

And why do we combine the custom error with the error of the `lookahead`? `lookahead.error()` should already tell you what it expects.

Reply:
I was thinking that the custom error message give slightly more context, but yes it is not needed here. I remove the custom error message.